### PR TITLE
Add workflow for updating core-example

### PR DESCRIPTION
**Pull request by nmlinaric**
## Description
Workflow that dispatches event to update core-example with latest chain bridge core version

## Related Issue Or Context
Closes: #168 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [ ] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

**Comment by mpetrun5** [comment]
Can you create also a receiver in chainbridge-core-example and test first from some local branch?
Github actions always needs a couple of tries :smile:

**Comment by nmlinaric** [comment]
> Can you create also a receiver in chainbridge-core-example and test first from some dev branch? Github actions always needs a couple of tries smile

Good idea for the test branch, PR already on core-example :smiley:

File: .github/workflows/update-core-example.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Update ChainBridge core-example
+
+on:
+  push:
+    branches:
+      - 'test-cd'
```
Comment (P1sar) [inline .github/workflows/update-core-example.yml:6]: main?

File: .github/workflows/update-core-example.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  update-core-example:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Trigger chainbridge-core-example update event
+        uses: peter-evans/repository-dispatch@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          # Dispatch event from current repository
+          repository: github.repository
+          event-type: update-core-example
```
Comment (P1sar) [inline .github/workflows/update-core-example.yml:18]: i would call it something like 'push-to-core-main-branch' to make it clear that this event is about new code on main branch in core repo

File: .github/workflows/update-core-example.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+jobs:
+  update-core-example:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Trigger chainbridge-core-example update event
+        uses: peter-evans/repository-dispatch@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          # Dispatch event from current repository
+          repository: github.repository
```
Comment (P1sar) [inline .github/workflows/update-core-example.yml:17]: isn't it should be repo name?

**Review by P1sar — CHANGES_REQUESTED** [review]

File: .github/workflows/update-core-example.yml (hunk 1 above)
Comment (nmlinaric) [reply .github/workflows/update-core-example.yml]: this is only temporary, until I test it

File: .github/workflows/update-core-example.yml (hunk 3 above)
Comment (nmlinaric) [reply .github/workflows/update-core-example.yml:17]: `github.repository` means the event dispatched from the current repository, that is `chainbridge-core`

**Review by MakMuftic — APPROVED** [review]

**Review by mpetrun5 — APPROVED** [review]

**Pull request merged**